### PR TITLE
Add `flake8-use-pathlib` lint check using ruff

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add `flake8-use-pathlib` lint check using ruff.

### Details

- Follow up of #5718 .

### CODE REVIEW

File: tests/core/test_polydata.py [hunk 1]
```diff
@@ -36,13 +36,13 @@ def cube_dense():
     return pv.Cube()
 
 
-test_path = os.path.dirname(os.path.abspath(__file__))
+test_path = Path(__file__).resolve().path
```
Comment (tkoyama010) [inline tests/core/test_polydata.py:39]:
```suggestion
test_path = Path(__file__).resolve().parent
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: pyvista/ext/plot_directive.py [hunk 2]
```diff
@@ -436,7 +436,7 @@ def render_figures(
                         continue
                     else:
                         image_file = ImageFile(output_dir, f"{output_base}_{i:02d}_{j:02d}.vtksz")
-                        with open(image_file.filename, "wb") as f:
+                        with Path(image_file.filename, "wb").open() as f:
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:439]:
```suggestion
                        with Path(image_file.filename).open("wb") as f:
```

File: pyvista/ext/plot_directive.py [hunk 3]
```diff
@@ -493,18 +493,21 @@ def run(arguments, content, options, state_machine, state, lineno):
...
         code = Path(source_file_name).read_text(encoding='utf-8')
-        output_base = os.path.basename(source_file_name)
+        output_base = Path(source_file_name).name
     else:
         source_file_name = rst_file
         code = textwrap.dedent("\n".join(map(str, content)))
         counter = document.attributes.get('_plot_counter', 0) + 1
         document.attributes['_plot_counter'] = counter
-        base, ext = os.path.splitext(os.path.basename(source_file_name))
+        path = Path(Path(source_file_name).name)
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:502]:
```suggestion
        path = Path(source_file_name)
```

File: pyvista/ext/plot_directive.py [hunk 4]
```diff
@@ -523,24 +526,22 @@ def run(arguments, content, options, state_machine, state, lineno):
...
     # see note in Python docs for warning about symbolic links on Windows.
     # need to compare source and dest paths at end
     build_dir = os.path.normpath(build_dir)
-    os.makedirs(build_dir, exist_ok=True)
+    Path(build_dir).mkdir(exist_ok=True)
 
     # output_dir: final location in the builder's directory
-    dest_dir = os.path.abspath(os.path.join(setup.app.builder.outdir, source_rel_dir))
-    os.makedirs(dest_dir, exist_ok=True)
+    dest_dir = str(Path(Path(setup.app.builder.outdir) / source_rel_dir).resolve())
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:540]:
```suggestion
    dest_dir = str((Path(setup.app.builder.outdir) / source_rel_dir).resolve())
```

File: tests/namespace/test_plotting_namespace.py [hunk 5]
```diff
@@ -1,12 +1,12 @@
...
-import pathlib
+from pathlib import Path
 
 import pytest
 
 from pyvista.core.errors import PyVistaDeprecationWarning
 
-namespace_data = pathlib.Path(__file__).parent / 'namespace-plotting.txt'
-with open(namespace_data) as f:
+namespace_data = str(Path(__file__).parent / 'namespace-plotting.txt')
```
Comment (tkoyama010) [inline tests/namespace/test_plotting_namespace.py:8]:
```suggestion
namespace_data = Path(__file__).parent / 'namespace-plotting.txt'
```

File: tests/namespace/test_plotting_namespace.py [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
+with Path(namespace_data).open() as f:
```
Comment (tkoyama010) [inline tests/namespace/test_plotting_namespace.py:9]:
```suggestion
with namespace_data.open() as f:
```

File: tests/namespace/test_public_namespace.py [hunk 7]
```diff
@@ -7,8 +8,8 @@
 # Use cleaned data to avoid things like `np`, `os`, etc
 # This prevents testing against things that are not intended
 # to be in the public namespace
-namespace_data = pathlib.Path(__file__).parent / 'namespace-top.txt'
-with open(namespace_data) as f:
+namespace_data = str(pathlib.Path(__file__).parent / 'namespace-top.txt')
```
Comment (tkoyama010) [inline tests/namespace/test_public_namespace.py:11]:
```suggestion
namespace_data = pathlib.Path(__file__).parent / 'namespace-top.txt'
```

File: tests/namespace/test_public_namespace.py [hunk 8]
```diff
@@ -7,8 +8,8 @@
...
+with Path(namespace_data).open() as f:
```
Comment (tkoyama010) [inline tests/namespace/test_public_namespace.py:12]:
```suggestion
with namespace_data.open() as f:
```

File: tests/namespace/test_utilities_namespace.py [hunk 9]
```diff
@@ -1,12 +1,13 @@
...
+from pathlib import Path
 
 import pytest
 
 from pyvista.core.errors import PyVistaDeprecationWarning
 
-namespace_data = pathlib.Path(__file__).parent / 'namespace-utilities.txt'
-with open(namespace_data) as f:
+namespace_data = str(pathlib.Path(__file__).parent / 'namespace-utilities.txt')
+with Path(namespace_data).open() as f:
```
Comment (tkoyama010) [inline tests/namespace/test_utilities_namespace.py:10]:
```suggestion
namespace_data = pathlib.Path(__file__).parent / 'namespace-utilities.txt'
with namespace_data.open() as f:
```

File: tests/plotting/test_scraper.py [hunk 10]
```diff
@@ -37,28 +36,28 @@ def test_scraper_with_app(tmpdir, monkeypatch, n_win=2):
...
+    target_file = str(Path(src_dir) / 'auto_examples' / 'sg.py')
     block = None
     block_vars = dict(
         image_path_iterator=iter(img_fnames),
         example_globals=dict(a=1),
         target_file=target_file,
     )
 
-    os.makedirs(op.dirname(img_fnames[0]))
+    Path(Path(img_fnames[0]).parent).mkdir(parents=True)
```
Comment (tkoyama010) [inline tests/plotting/test_scraper.py:53]:
```suggestion
    Path(img_fnames[0]).parent.mkdir(parents=True)
```

File: tests/plotting/test_scraper.py [hunk 11]
```diff
@@ -87,36 +86,36 @@ def test_scraper(tmpdir, monkeypatch, n_win, scraper_type):
...
+    target_file = str(Path(src_dir) / 'auto_examples' / 'sg.py')
     block = ("empty_block", "", 0)
     block_vars = dict(
         image_path_iterator=iter(img_fnames),
         example_globals=dict(a=1, PYVISTA_GALLERY_FORCE_STATIC_IN_DOCUMENT=True),
         target_file=target_file,
     )
 
-    os.makedirs(op.dirname(img_fnames[0]))
+    Path(Path(img_fnames[0]).parent).mkdir(parents=True)
```
Comment (tkoyama010) [inline tests/plotting/test_scraper.py:109]:
```suggestion
    Path(img_fnames[0]).parent.mkdir(parents=True)
```

File: tests/plotting/test_scraper.py [hunk 12]
```diff
@@ -127,19 +126,19 @@ def test_scraper_raise(tmpdir):
...
     block = None
     block_vars = dict(
         image_path_iterator=(img for img in [img_fname]),
         example_globals=dict(a=1),
         target_file=target_file,
     )
-    os.makedirs(op.dirname(img_fname))
-    assert not os.path.isfile(img_fname)
-    os.makedirs(out_dir)
+    Path(str(Path(img_fname).parent)).mkdir(parents=True)
```
Comment (tkoyama010) [inline tests/plotting/test_scraper.py:139]:
```suggestion
    Path(img_fname).parent.mkdir(parents=True)
```

**Comment by tkoyama010** [comment]
pre-commit.ci autofix

File: pyvista/ext/plot_directive.py [hunk 13]
```diff
@@ -493,18 +493,21 @@ def run(arguments, content, options, state_machine, state, lineno):
...
-        output_base = os.path.basename(source_file_name)
+        output_base = Path(source_file_name).name
     else:
         source_file_name = rst_file
         code = textwrap.dedent("\n".join(map(str, content)))
         counter = document.attributes.get('_plot_counter', 0) + 1
         document.attributes['_plot_counter'] = counter
-        base, ext = os.path.splitext(os.path.basename(source_file_name))
+        path = Path(source_file_name)
+        base = str(path.parent / path.stem)
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:503]:
```suggestion
        base = str(path.parent / path.stem)
        ext = path.suffix
```

File: pyvista/ext/plot_directive.py [hunk 14]
```diff
@@ -493,18 +493,21 @@ def run(arguments, content, options, state_machine, state, lineno):
...
-        output_base = os.path.basename(source_file_name)
+        output_base = Path(source_file_name).name
     else:
         source_file_name = rst_file
         code = textwrap.dedent("\n".join(map(str, content)))
         counter = document.attributes.get('_plot_counter', 0) + 1
         document.attributes['_plot_counter'] = counter
-        base, ext = os.path.splitext(os.path.basename(source_file_name))
+        path = Path(Path(source_file_name).name)
+        base = path.parent / path.stem
```
Comment (tkoyama010) [inline pyvista/ext/plot_directive.py:503]:
```suggestion
        base = str(path.parent / path.stem)
```

**Comment by tkoyama010** [comment]
@pyvista-bot LGTM

File: pyvista/plotting/themes.py [hunk 15]
```diff
@@ -1388,7 +1389,7 @@ def __init__(self):
         service = os.environ.get('JUPYTERHUB_SERVICE_PREFIX', '')
         prefix = os.environ.get('PYVISTA_TRAME_SERVER_PROXY_PREFIX', '/proxy/')
         if service and not prefix.startswith('http'):  # pragma: no cover
-            self._server_proxy_prefix = os.path.join(service, prefix.lstrip('/'))
+            self._server_proxy_prefix = str(Path(service) / prefix.lstrip('/'))
```
Comment (banesullivan) [inline pyvista/plotting/themes.py:1392]: This was a breaking change

Comment (banesullivan) [reply pyvista/plotting/themes.py:1392]: Pathlib strips the trailing slash which we need here

File: pyvista/plotting/themes.py [hunk 16]
```diff
@@ -3058,7 +3059,7 @@ def save(self, filename: str) -> None:
         data = self.to_dict()
         # functions are not serializable
         del data["before_close_callback"]
-        with open(filename, 'w') as f:
+        with Path(filename).open('w') as f:
```
Comment (banesullivan) [inline pyvista/plotting/themes.py:3062]: I'm really not a fan of these changes... `open(filename, 'w')` has be the pattern for ages. I don't love seeing this well established convention replaced.

**Pull request merged**